src/components/pages/collectRecordFormPages/HabitatComplexityForm/HabitatComplexityForm.js

```diff
@@ -114,8 +114,12 @@ const HabitatComplexityForm = ({ isNewRecord }) => {
     )
   }, [collectRecordBeingEdited, getPersistedUnsavedFormikData])
 
-  const handleCollectRecordChange = (updatedCollectRecord) =>
+  const handleCollectRecordChange = (updatedCollectRecord) => {
+    // maybe rename this function. Its pretty vague...
+    // need to recompute record name for sub nav node for all CRs
+    // setSubNavNode(recordNameForSubNode)
```

Comment:
in the `updatedCollectRecord` response, we have  `data.sample_event.site` with a long id string, I *pretty sure* I remember seeing a function that decodes this into a name, but don't remember which one @mmnoo

Reply:
is the the site id ? why don't we also include the actual names in the response

Reply:
That is a good question. I dont know. It would make the front end code cheaper to build/maintain. And probably faster (avoiding reading IDB extra times to join data)..

I think the goal at the time was to change the api minimally from v1... and it also had to be backward compatible. Now that we are killing v1, maybe there are opportunities moving forward for new features at least.

Reply:
yes its site id

Reply:
But that setSubNode function should do it for you magically if make this code above reusable here

```
              const recordNameForSubNode =
                !isNewRecord && collectRecordResponse
                  ? getRecordSubNavNodeInfo(
                      collectRecordResponse.data,
                      sitesResponse,
                      collectRecordResponse.data.protocol,
                    )
                  : { name: language.protocolTitles.habitatcomplexity }
                  ```

Reply:
@mmnoo so setSubNode uses `getRecordSubNavNodeInfo` which requires 3 params -` collectRecordResponse`, `sites`, and `protocol`.  

we get `sites` from  `databaseSwitchboardInstance.getSitesWithoutOfflineDeleted` . it looks like we would have to call this again within `handleCollectRecordChange` to get fresh sites data incase one of the names changes, however, we would have to somehow check if that was the case. in this case,  it feels like we are doing too much just to get the site name, which is included in the `sample_event` info, however its as an id string

Reply:
I think now that we are killing v1, it would be good to start implementing some of those changes. this is heavy work for the front end

Reply:
ooooooooo